test/unit/preprocessing/linear_dep_remover/test_correlation_remover_efficiency.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable
+
+import numpy as np
+import pandas as pd
+import pytest
+from numpy.typing import NDArray
+from scipy.stats import wilcoxon
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+from fairlearn.datasets import fetch_adult
+from fairlearn.metrics import (
+    demographic_parity_difference,
+    equal_opportunity_difference,
+    equalized_odds_difference,
+)
+from fairlearn.preprocessing import CorrelationRemover
+
+
+@dataclass(frozen=True)
+class FairnessComparisonData:
+    sensitive_feature: pd.Series
+    y_true: pd.Series
+    y_pred_with_mitigation: NDArray
+    y_pred_without_mitigation: NDArray
+
+    def bootstrap(self, seed: int) -> FairnessComparisonData:
+        sample_indices = np.random.RandomState(seed=seed).choice(
+            len(self.sensitive_feature), len(self.sensitive_feature), replace=True
+        )
+        sampled_sensitive_feature = self.sensitive_feature.iloc[sample_indices]
+        sampled_y_true = self.y_true.iloc[sample_indices]
+        sampled_y_pred_with_mitigation = self.y_pred_with_mitigation[sample_indices]
+        sampled_y_pred_without_mitigation = self.y_pred_without_mitigation[sample_indices]
+
+        return FairnessComparisonData(
+            sensitive_feature=sampled_sensitive_feature,
+            y_true=sampled_y_true,
+            y_pred_with_mitigation=sampled_y_pred_with_mitigation,
+            y_pred_without_mitigation=sampled_y_pred_without_mitigation,
+        )
+
+
+def _get_adult_fairness_comparison_data() -> FairnessComparisonData:
+    features_to_keep = ["fnlwgt", "capital-gain", "capital-loss", "hours-per-week", "age"]
+    senstive_feature_id = "race_White"
+    raw_data = fetch_adult().frame
+    data, target = (
+        raw_data[features_to_keep + ["race"]],
+        raw_data["class"] == ">50K",
+    )
+    data = pd.get_dummies(data)[features_to_keep + [senstive_feature_id]]
+    X_train, X_test, y_train, y_test = train_test_split(
+        data, target, test_size=0.2, random_state=0
+    )
+
+    cr = CorrelationRemover(sensitive_feature_ids=[senstive_feature_id])
+    cr.fit(X_train)
+    X_train_transformed = cr.transform(X_train)
+    X_test_transformed = cr.transform(X_test)
+
+    estimator_with_correlation = LogisticRegression(solver="liblinear")
+    estimator_without_correlation = LogisticRegression(solver="liblinear")
+    estimator_with_correlation.fit(X_train, y_train)
+    estimator_without_correlation.fit(X_train_transformed, y_train)
+
+    return FairnessComparisonData(
+        sensitive_feature=X_test[senstive_feature_id],
+        y_true=y_test,
+        y_pred_without_mitigation=estimator_with_correlation.predict(X_test),
+        y_pred_with_mitigation=estimator_without_correlation.predict(X_test_transformed),
+    )
+
+
+ADULT_FAIRNESS_COMPARISON_DATA = _get_adult_fairness_comparison_data()
```

Comment:
I don't think we should be fitting in the get method here. In effect, this runs the `fit` during `import` time, which I think is not a good idea. I'd be happy with the function to return the data which is then to be used inside the test methods.

Reply:
Thanks @adrinjalali that's a great catch indeed :100: 

What I did to fix it is:
- pass the data preparation function to the test function, so that the fit (of both the transformer and the estimators) happens within it. It also keeps the test extendible: one just has to write a new function that returns a `FairnessComparisonData` and add it to the test parameters.
- to avoid unnecessarily repeating the fitting, I now compare all the metrics in one go. The test is still easily extendible in that regard also.